x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/shared/FrozenCacheService.java

```diff
@@ -117,12 +118,23 @@ public void validate(final ByteSizeValue value, final Map<Setting<?>, Object> se
                             roles.stream().map(DiscoveryNodeRole::roleName).collect(Collectors.joining(","))
                         );
                     }
+
+                    @SuppressWarnings("unchecked")
+                    final List<String> dataPaths = (List<String>) settings.get(Environment.PATH_DATA_SETTING);
+                    if (dataPaths.size() > 1) {
+                        throw new SettingsException(
+                            "setting [{}={}] is not permitted on nodes with multiple data paths [{}]",
```

Comment:
How does this interact with the other PR (default frozen cache size) when you have multiple data paths defined and a dedicated frozen node defined, but have not explicitly configured a cache size?

Reply:
It is my intention to make that case fail once I merge the two together.